### ISSUE TITLE
rig-remote dumps stack trace when no config folder is present (was: Not working on OS X 10.11.6)

### ISSUE THREAD

**Issue by stuzog** [Marzona/rig-remote#82]
On an iMac 7,1 running OS X 10.11.6 I get this with rig-remote-2.0:

```
bash$ pwd
/Applications/SDR/rig-remote-2.0
bash$ ./test.sh
usage: pytest.py [options] [file_or_dir] [file_or_dir] [...]
pytest.py: error: unrecognized arguments: --cov-report --cov-config /Users/Stuart/.coveragerc --cov modules ./test
  inifile: None
  rootdir: /Applications/SDR/rig-remote-2.0
bash$ 
```

I made a file `rig-bookmarks.csv` formatted according to the README but it didn't find it, either in `rig-remote` or `rig-remote-2.0`:

```
bash$ pwd
/Applications/SDR/rig-remote
bash$ python ./rig-remote.py
Traceback (most recent call last):
  File "./rig-remote.py", line 147, in <module>
    elif ac.config['bookmark_filename'] == 'noname':
KeyError: 'bookmark_filename'

bash$ cd ../rig-remote-2.0/
bash$ python ./rig-remote.py
Traceback (most recent call last):
  File "./rig-remote.py", line 103, in <module>
    app = RigRemote(root, ac)
  File "/Applications/SDR/rig-remote-2.0/modules/ui.py", line 297, in __init__
    self.bookmark("load", ",")
  File "/Applications/SDR/rig-remote-2.0/modules/ui.py", line 1032, in bookmark
    line[BM.freq] = self._frequency_pp(line[BM.freq])
  File "/Applications/SDR/rig-remote-2.0/modules/ui.py", line 1440, in _frequency_pp
    return '{:,}'.format(int(re.sub("[^0-9]", '', frequency)))
ValueError: invalid literal for int() with base 10: ''
bash$ 
```

**Comment by Marzona** [comment]
there is a new release coming out soon from master branch. In master branch i fixed some bugs already. I am setting up a better testing environment on OSs different than Linux (the platform on which I develop rig-remote).

Meanwhile can you checkout the code now in Master and check if you still have the behavior you have described here?

**Comment by stuzog** [comment]
Downloaded latest but build did not start, giving an error in reference to the bookmarks file. I've had this problem all along and have not been able get r-r to recognise the bookmarks `rig-bookmarks.csv` file I created and placed in the r-r directory:

```
bash$ cd rig-remote/
bash$ ls -al
total 72
drwxr-xr-x  14 Stuart  admin   476  2 Jan 12:51 .
drwxr-xr-x  17 Stuart  admin   578  2 Jan 12:47 ..
drwxr-xr-x  13 Stuart  admin   442  2 Jan 12:47 .git
-rw-r--r--   1 Stuart  admin  1115  2 Jan 12:47 LICENSE
-rw-r--r--   1 Stuart  admin    18  2 Jan 12:47 MANIFEST.in
-rw-r--r--   1 Stuart  admin  3935  2 Jan 12:47 README.md
-rwxr-xr-x   1 Stuart  admin   526  2 Jan 12:47 release.sh
-rw-r--r--@  1 Stuart  admin   412  1 Jan 18:00 rig-bookmarks.csv
-rwxr-xr-x   1 Stuart  admin  5179  2 Jan 12:47 rig-remote.py
drwxr-xr-x  26 Stuart  admin   884  2 Jan 12:49 rig_remote
drwxr-xr-x   3 Stuart  admin   102  2 Jan 12:47 screenshots
-rw-r--r--   1 Stuart  admin   125  2 Jan 12:47 setup.cfg
-rw-r--r--   1 Stuart  admin  2213  2 Jan 12:47 setup.py
drwxr-xr-x  16 Stuart  admin   544  2 Jan 12:47 test
bash$ python ./rig-remote.py
Traceback (most recent call last):
  File "./rig-remote.py", line 147, in <module>
    elif ac.config['bookmark_filename'] == 'noname':
KeyError: 'bookmark_filename'
bash$
```

**Comment by stuzog** [comment]
BTW, I noticed `rig-remote/test/test_bookmarls.py` -- should that not be `test_bookmarks.py`?

**Comment by Marzona** [comment]
in 4ef68e506ee8d88233e5745a60a42e02dd229b68 I fixed the name of the test file.

Despite being a typo in the file name, it is a file containing unit tests, so it is not affecting the tool itself. That file is not included anywhere in the project.

**Comment by Marzona** [comment]
ok, I managed to reproduce the problem. I'll work on that for the release 3 of rig-remote.

Thanks for all the info. I'll update this issue as soon as I have updates.

**Comment by Marzona** [comment]
The branch for the work on this issue is https://github.com/Marzona/rig-remote/tree/issues-82

**Comment by stuzog** [comment]
OK, thanks Simone.

NOTE: link in `README.md;` 
<https://github.com/Marzona/rig-remote/blob/master/screenshots/rig-remote-fork.png> 
is broken.

**Comment by Marzona** [comment]
link fixed

**Comment by stuzog** [comment]
Also in README.md:

STARTS CLIP->

```
If you are using Windows you just need to double-click the 
```rig-remote.py``` file (as the ```.py``` file type is most likely 
already bound with ```python``` executable). If you want to get rid of 
the anoying command-line that is always running in background you may 
rename ```rig-remote.py``` to ```rig-remote.pyw``` and Windows will use 
the ```pythonw``` executable instead (which does not need the command-line).

This software consists of two files and two folder:
=======
- rig-remote.py
- rig-bookmarks.csv (the bookmark database)
- modules: python modules
- tests: unit tests

The file ```rig-bookmarks.csv``` consists on a standard comma-separated 
values file. For reference, the following example file is provided:
```

ENDS CLIP->

should be:

STARTS CLIP->

If you are using Windows you just need to double-click the 
`rig-remote.py` file (as the `.py` file type is most likely already 
bound with `python` executable). If you want to get rid of the anoying 
command-line that is always running in background you may rename 
`rig-remote.py` to `rig-remote.pyw` and Windows will use the `pythonw` 
executable instead (which does not need the command-line).

This software consists of two files and two folder:
=======
- rig-remote.py
- rig-bookmarks.csv (the bookmark database)
- modules: python modules
- tests: unit tests

The file `rig-bookmarks.csv` consists of a standard comma-separated 
values file. For reference, the following example file is provided:

ENDS CLIP->

(In-line code in Markdown only requires one backtick; code blocks take 
three @beginning and @end on a separate line, separated from 
preceeding/following text by a blank line.)

Thanks for your prompt responses!

Stuart

**Comment by stuzog** [comment]
Hi Simone,

Here's an edit of the README.md file with corrected markdown and (I 
hope) slightly improved language!

Stuart

**Comment by Marzona** [comment]
Thanks

**Comment by stuzog** [comment]
Hi Simone,

I've been making progress with getting rig-remote to run on OS X, using 
this invocation in the r-r directory in Terminal:

`python rig-remote.py --bookmarks rig-bookmarks.csv --log rig-logfile.txt`

The r-r window opens with a floating python warning window:

*Invalid value in Bookmark #0. Skipping...*

Clicking OK just closes and immediately reopens it, so I have to close 
the Terminal window to stop r-r.

I guess there needs to be a suitable config file, and documentation of 
what ca go in it.

Stuart

Simone Marzona wrote:

**Comment by Marzona** [comment]
the logic in rig-remote should create a default config file if no config file is found, and this logic needs to be fixed.
https://github.com/Marzona/rig-remote/issues/84 should capture the configuration file documentation

**Comment by Marzona** [comment]
https://github.com/Marzona/rig-remote/issues/85 should capture the above suggestion you provided

**Comment by Marzona** [comment]
https://github.com/Marzona/rig-remote/issues/86 has been created for re-factoring the part of the code that handles the configuration of the tool (defaults, save/load at all).

**Comment by Marzona** [comment]
The reported behavior should be fixed in commit: 9d637116bb358eac71b2e8ff59d3d14f98904a63 now in branch https://github.com/Marzona/rig-remote/tree/issues-82.

I am going to test this codechange on a  "clean" machines before merging into master branch.

**Comment by Marzona** [comment]
test done on clean installs of Win, OSX and Linux. The tool loads as expected.
